src/modules/filter-drawer/filter-drawer.tsx

```diff
@@ -0,0 +1,349 @@
+
+      // Content Wrapper
+      Animated.timing(contentWrapperTranslateX, {
+        toValue: -windowWidth,
+        useNativeDriver: true,
+        duration: 300,
+      }),
+    ]).start();
+  };
+
+  // On Close Modal
+  const onCloseModal = () => {
+    // @ts-ignore
+    modalRef?.current?.close?.();
+
+    // Call callback function
+    onClose();
+  };
+
+  React.useEffect(() => {
+    if (shouldShowSecondaryContent) {
+      onShowSecondaryContent();
+    } else {
+      onShowPrimaryContent();
+    }
+  }, [shouldShowSecondaryContent]);
+
+  const AnimatedWrapper = Animated.createAnimatedComponent(ContentWrapper);
+  const AnimatedCloseWrapper =
+    Animated.createAnimatedComponent(CloseButtonWrapper);
+
+  const modalStyle = {margin: 0};
+
+  return (
+    <Modal
+      ref={modalRef}
+      isVisible={isOpen}
+      onModalHide={onClose}
+      hideModalContentWhileAnimating
+      useNativeDriverForBackdrop
+      style={modalStyle}
+      useNativeDriver
+      propagateSwipe>
+      <Wrapper testID="drawer-wrapper">
+        {/* Sections */}
+        <SectionWrapper>
+          {/* Content */}
+          <AnimatedWrapper
+            style={{transform: [{translateX: contentWrapperTranslateX}]}}>
+            {/* Primary Content */}
+            <PrimaryContentWrapper testID="drawer-primary-content">
+              <ScrollView showsVerticalScrollIndicator={false}>
+                {/* Header */}
+                <Header.Wrapper>
+                  <Animated.View
+                    style={{
+                      opacity: headerContentOpacity,
+                      transform: [{translateX: headerContentTranslateX}],
+                    }}>
+                    <Header.Title variant="h2">Filters</Header.Title>
+
+                    {shouldShowClearFiltersButton && (
+                      <Header.Action testID="drawer-clear-filters-action">
+                        <Link
+                          onPress={onClearFilters}
+                          label="Clear Filters"
+                          small
+                        />
+                      </Header.Action>
+                    )}
+                  </Animated.View>
+                </Header.Wrapper>
+
+                {primaryChildren}
+              </ScrollView>
+
+              {/* Footer */}
+              <FooterWrapper>
+                <SafeAreaView>
+                  <ButtonGroup>
+                    {shouldShowCancelButton && (
+                      <Button
+                        color="secondary"
+                        appearance="outline"
+                        testID="drawer-cancel-filters-button"
+                        onPress={onCancel}>
+                        Cancel
+                      </Button>
+                    )}
+
+                    <Button
+                      color="success"
+                      testID="drawer-apply-filters-button"
+                      onPress={onApplyFilters}>
+                      Apply Filters
```

Comment:
'Apply filters' button title should be dynamic and should have visibility boolean flag. So consumer can show/hide if filters selected or not.